File: 202003091636 - ee1122 (C# - Dershane Otomasyon)/example-code/dershane otomasyonları/dershane otomasyonları/c#/dershane_otomasyon/dershane_otomasyon/Form4.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data.OleDb;

namespace dershane_otomasyon
{
    public partial class Form4 : Form
    {
        public Form4()
        {
            InitializeComponent();
        }
        OleDbConnection baglan = new OleDbConnection("Provider=Microsoft.ace.OleDb.12.0;Data Source=dershane.accdb");
        OleDbCommand kmt = new OleDbCommand();
        DataTable tablo = new DataTable();
        OleDbDataAdapter adtr = new OleDbDataAdapter();
        private void GroupBox1_Enter(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {
            Form2 FRM2 = new Form2();
            FRM2.Show();
            this.Hide();
        }

        private void dataGridView1_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            TextBox1.Text = dataGridView1.Rows[e.RowIndex].Cells[1].Value.ToString();
            TextBox2.Text = dataGridView1.Rows[e.RowIndex].Cells[2].Value.ToString();
            TextBox3.Text = dataGridView1.Rows[e.RowIndex].Cells[3].Value.ToString();
            TextBox4.Text = dataGridView1.Rows[e.RowIndex].Cells[4].Value.ToString();
            TextBox5.Text = dataGridView1.Rows[e.RowIndex].Cells[5].Value.ToString();
           
           
        }
        public void listele()
        {
            TextBox1.Text = "";
            TextBox2.Text = "";
            TextBox3.Text = "";
            TextBox4.Text = "";
            TextBox5.Text = "";
            tablo.Clear();
            adtr = new OleDbDataAdapter(" Select * from personel ", baglan);
            adtr.Fill(tablo);
            dataGridView1.DataSource = tablo;
        }
        private void button2_Click(object sender, EventArgs e)
        {
            if (baglan.State == ConnectionState.Open)
            {

                baglan.Close();
            }
            baglan.Open();



            kmt = new OleDbCommand("INSERT INTO personel(adi,soyadi,tel,pers_maas,brans) VALUES('" + TextBox1.Text + "','" + TextBox2.Text + "','" + TextBox3.Text + "','" + TextBox4.Text + "','" + TextBox5.Text + "')", baglan);
            kmt.ExecuteNonQuery();
            baglan.Close();
            MessageBox.Show("Kayıt Başarılı");
            listele();
        }

        private void Form4_Load(object sender, EventArgs e)
        {
            listele();
        }

        private void button3_Click(object sender, EventArgs e)
        {

            baglan.Open();
            kmt = new OleDbCommand("Delete from personel WHERE adi = '" + TextBox1.Text + "'", baglan);
            kmt.ExecuteNonQuery();
            kmt.Dispose();
            baglan.Close();
            MessageBox.Show("İşleminiz başarılı");
            listele();
        }

        private void button4_Click(object sender, EventArgs e)
        {
            baglan.Open();
            kmt = new OleDbCommand("UPDATE personel SET adi='" + TextBox1.Text + "',soyadi='" + TextBox2.Text + "',tel='" + TextBox3.Text + "',pers_maas='" + TextBox4.Text + "',brans='"+TextBox5.Text+"'", baglan);
            kmt.ExecuteNonQuery();
            baglan.Close();
            MessageBox.Show("İşleminiz başarılı");
            listele();
        }

        private void button5_Click(object sender, EventArgs e)
        {
            baglan.Open();
            if (textBox6.Text.Trim() != "")
            {
                adtr.SelectCommand.CommandText = " Select * From personel" +
                     " where(adi='" + textBox6.Text + "' )";
                tablo.Clear();
                adtr.Fill(tablo);
                baglan.Close();
            }
        }
    }
}

File: 202003091636 - ee1122 (C# - Dershane Otomasyon)/example-code/dershane otomasyonları/dershane otomasyonları/vb/Dershane Projeleri/DershaneOtomasyon/DershaneOtomasyon/Form4.vb
Imports System.Data

Imports System.Data.OleDb

Public Class Form4
    Public table As New DataTable




    Public Sub ogrenci()

        table.Clear()

        Dim adtr As New OleDbDataAdapter(" Select * from ogr ", Form2.bag)

        adtr.Fill(table)

        dataGridView1.DataSource = table

    End Sub

    Private Sub Form4_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ogrenci()

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Form2.bag.Open()

        Form2.kmt.Connection = Form2.bag

        Form2.kmt.CommandText = "DELETE FROM ogr WHERE numarasi = '" + textBox1.Text & "'"

        Form2.kmt.ExecuteNonQuery()

        Form2.bag.Close()




        ogrenci()



        MsgBox("Silme işlemi gerçekleştirildi !")
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Form2.bag.Open()

        Form2.kmt.Connection = Form2.bag

        Form2.kmt.CommandText = (((((((((((("UPDATE ogr SET numarasi='" + TextBox1.Text & "' , adi='") + TextBox2.Text & "',soyadi='") + TextBox3.Text & "', sinifi='") + TextBox4.Text & "',cinsiyeti='") + TextBox5.Text & "',bölümü='") + TextBox6.Text & "',dali ='") + TextBox7.Text & "',ev_adresi='") + TextBox8.Text & "',posta_kodu='") + TextBox9.Text & "',semt='") + TextBox10.Text & "', sehir='") + TextBox11.Text & "',telefon='") + TextBox12.Text & "',cep_telefon='") + TextBox13.Text & "' where numarasi='" & TextBox1.Text & "'"

        Form2.kmt.ExecuteNonQuery()

        Form2.kmt.Dispose()

        Form2.bag.Close()



        ogrenci()



        MsgBox("Güncelleme işlemi gerçekleştirildi !")
    End Sub

    Private Sub Label14_Click(sender As Object, e As EventArgs) Handles Label14.Click
        Form2.Show()
        Me.Hide()

    End Sub

    Private Sub Label15_Click(sender As Object, e As EventArgs) Handles Label15.Click
        Application.Exit()
    End Sub
End Class